File: 序号/589.N叉树的前序遍历.kt

/**
 * Definition for a Node.
 * class Node(var `val`: Int) {
 *     var children: List<Node?> = listOf()
 * }
 */

class Solution {
	fun preorder(root: Node?): List<Int> {
		if (root == null) return listOf()
		val ans = mutableListOf<Int>()
		ans.add(root.`val`)
		root.children.forEach {
			ans.addAll(preorder(it))
		}
		return ans
	}
}

File: 序号/589. N 叉树的前序遍历.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

# 24.02.18
class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        ans=[]
        def dfs(root):
            if not root:return
            ans.append(root.val)
            for child in root.children:
                dfs(child)
        dfs(root)
        return ans
# class Solution:
#     def preorder(self, root: 'Node') -> List[int]:
        
#         l=[]
#         if not root:return l
#         def dfs(node):
#             if node:
#                 l.append(node.val)
#                 if node.children:
#                     for c in node.children:
#                         dfs(c)
#             else :
#                 l.append(None)
#         dfs(root)
#         return l
